src/main/screen_pause.cpp

```diff
@@ -223,22 +243,20 @@ void Init(bool LegacyPause)
     else
     {
         s_items.push_back(MenuItem{"CONTINUE", s_Continue});
-
+        
         if(g_compatibility.allow_drop_add && !LegacyPause)
             s_items.push_back(MenuItem{"DROP/ADD PLAYERS", s_DropAddScreen});
 
         if(g_config.enter_cheats_menu_item && !LegacyPause)
             s_items.push_back(MenuItem{"ENTER CHEAT", s_CheatScreen});
-
-        if(CanSave)
-        {
-            s_items.push_back(MenuItem{"SAVE & CONTINUE", s_SaveAndContinue});
-            s_items.push_back(MenuItem{"SAVE & QUIT", s_Quit});
-        }
-        else
+        
+        if(IsOnLevel)
         {
-            s_items.push_back(MenuItem{"QUIT", s_Quit});
+            s_items.push_back(MenuItem{"RETURN TO MAP/HUB", s_ExitLevel});
         }
+        
+        s_items.push_back(MenuItem{"SAVE & CONTINUE", s_SaveAndContinue});
+        s_items.push_back(MenuItem{"SAVE & QUIT", s_Quit});
```

Comment:
Dude, you shall NOT change this behaviour mandatorily, you should implement the `compat.ini` option (look at `copmat.h` and `compat.cpp`) to enable/disable this. It's critical for speed-runners, they must expect the same behaviour as the original SMBX, otherwise, it's a time-winning function. So:
- add the boolean field that enables the direct quit to the world map feature
- look at conditions of compatibility modes (there are modes: Modern, SMBX2, and SMBX13). So, when the game is running under these compat modes, you should ensure it will be FALSE by default when a certain compat mode enforces it.

Also, you shall NOT remove the "no-save" punish for cheating. Yet. Or again, add the `compat.ini` option that declares this behaviour, it's significantly important.